scrapper.py

```diff
@@ -23,75 +36,310 @@ class IncorrectNumberOfArticlesError(Exception):
 
 class UnknownConfigError(Exception):
     """
-    Most general error
+    Custom error
+    """
+
+
+class BadStatusCode(Exception):
+    """
+    Custom error
     """
 
 
 class Crawler:
     """
     Crawler implementation
     """
-    def __init__(self, seed_urls: list, max_articles: int):
-        pass
+    headers = {'user-agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 '
+                             '(KHTML, like Gecko) Chrome/88.0.4324.182 Safari/537.36'}
+
+    def __init__(self,
+                 seed_urls: list,
+                 max_articles: int = None,
+                 max_articles_per_seed: int = None):
+        self.seed_urls = seed_urls  # to parse
+        self.max_articles = max_articles
+        self.max_articles_per_seed = max_articles_per_seed
+        self.seen_urls = set()
+        self.urls = []  # articles urls
 
     @staticmethod
-    def _extract_url(article_bs):
-        pass
+    def _extract_url(article_bs, article_urls, seen_urls):
```

Comment:
I do not remember that it was approved to change the interface. Can you point me where it was allowed? or align with the requirements

Reply:
sorry, got the previous interface back